Gordon360/ApiControllers/HousingController.cs

```diff
@@ -206,24 +210,30 @@ public IHttpActionResult EditApplication([FromBody] ApartmentAppViewModel apartm
                 throw new BadInputException() { ExceptionMessage = errors };
             }
             //get token data from context, username is the username of current logged in person
-            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
-            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+            ClaimsPrincipal authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            string username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
 
-            string editorId = _accountService.GetAccountByUsername(username).GordonID;
+            string userID = _accountService.GetAccountByUsername(username).GordonID;
 
-            string sessionId = Helpers.GetCurrentSession().SessionCode;
+            string sessionID = Helpers.GetCurrentSession().SessionCode;
+
+            if (applicationID == -1)
+            {
+                applicationID = applicationDetails.AprtAppID; // The apartAppID is set to -1 if an existing application ID was not yet known by the frontend
```

Comment:
This is more of a backwards-compatibility thing, so that it will try to get it from the JSON of the FromBody if it somehow got this far......

I was trying to make it so it could still be compatible with the old branches that haven't been updated to match, but I see now that it was a bad idea for several other reasons as well

Reply:
You were not misunderstanding this, this was a legitimate mistake and you were right to be confused by it

Reply:
Oh, good!